graphql/lessonplan/mutations.ts

```diff
@@ -0,0 +1,219 @@
+import { Resolver, Mutation, Arg, Authorized, Ctx, InputType, Field, Int, ObjectType, registerEnumType } from 'type-graphql';
+import { MaxLength, ArrayMaxSize, validateOrReject } from 'class-validator';
+import { Role } from '../authorizations';
+import { GraphQLContext } from '../context';
+import { getLogger } from '../../common/logger/logger';
+import { generateLessonPlan } from '../../common/lessonplan/generate';
+import { course_subject_enum, pupil_state_enum, school_schooltype_enum } from '@prisma/client';
+import { ApolloError, UserInputError } from 'apollo-server-express';
+
+const logger = getLogger(`LessonPlan Mutations`);
+
+// Maximum prompt length (in characters)
+const MAX_PROMPT_LENGTH = 10000;
+
+// Maximum number of files
+const MAX_FILES = 10;
+
+// Register the course_subject_enum as a GraphQL enum
+registerEnumType(course_subject_enum, {
```

Comment:
Move to https://github.com/corona-school/backend/blob/master/graphql/types/enums.ts - there we already added this for other types (unfortunately this is missing in prisma-graphql)